daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/JsonLedgerClient.scala

```diff
@@ -343,38 +349,13 @@ class JsonLedgerClient(
           )
       }
     } yield result
-  }
-
-  override def submitMustFail(
-      actAs: OneAnd[Set, Ref.Party],
-      readAs: Set[Ref.Party],
-      disclosures: List[Disclosure],
-      commands: List[command.ApiCommand],
-      optLocation: Option[Location],
-  )(implicit ec: ExecutionContext, mat: Materializer) = {
-    submit(actAs, readAs, disclosures, commands, optLocation).map {
-      case Right(_) => Left(())
-      case Left(_) => Right(())
+    // Json api doesn't give back a SubmitError or a transaction tree.
+    resultF.map {
+      case Left(err) => Left(ScriptLedgerClient.SubmitFailure(err, None))
+      case Right(results) => Right((results, None))
```

Comment:
the optional transaction tree bugs me a bit since it worsens the API for everyone, even if they don't use the json client. Do you know who/what code path does use the json client?

Reply:
Ah, you're in luck! we got permission to remove the json client, so I can fix this up in a later PR :)

Reply:
yeet!

Reply:
[#18033](https://github.com/digital-asset/daml/issues/18033)